operator/e2e/tests/cert_manager_test.go

```diff
@@ -0,0 +1,247 @@
+// /*
+// Copyright 2025 The Grove Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+//go:build e2e
+
+package tests
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ai-dynamo/grove/operator/e2e"
+	"github.com/ai-dynamo/grove/operator/e2e/setup"
+	"github.com/ai-dynamo/grove/operator/e2e/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	certManagerIssuerYAML = `
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-issuer
+spec:
+  selfSigned: {}
+`
+	certManagerCertificateYAML = `
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: grove-webhook-server-cert
+  namespace: grove-system
+spec:
+  secretName: grove-webhook-server-cert
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  issuerRef:
+    name: selfsigned-issuer
+    kind: ClusterIssuer
+  dnsNames:
+  - grove-operator
+  - grove-operator.grove-system
+  - grove-operator.grove-system.svc
+  - grove-operator.grove-system.svc.cluster.local
+`
+)
+
+func TestCertManagerIntegration(t *testing.T) {
```

Comment:
This tests going from autoProvision => cert manager. Let's create another test that starts with cert manager and goes to auto provsion, so we're testing both paths.  Were ever possible, don't forget to write helper functions in order to keep things DRY: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself